src/core/linter-rules/no-unused-dfns.js

```diff
@@ -0,0 +1,54 @@
+// @ts-check
+/**
+ * Linter rule "no-unused-dfns".
+ *
+ * Complains if an internal/un-exported definitions is not linked to.
+ */
+import {
+  docLink,
+  getIntlData,
+  norm,
+  showError,
+  showWarning,
+} from "../utils.js";
+
+const ruleName = "no-unused-dfns";
+export const name = "core/linter-rules/no-unused-dfns";
+
+const localizationStrings = {
+  en: {
+    msg(text) {
+      return `Found definition for "${text}", but nothing links to it. This is usually a spec bug!`;
+    },
+    get hint() {
+      return docLink`Either remove the definition, or add a \`"lint-ignore"\` CSS class. Or did you want to ${"[export|#data-export]"} it? If yes, add the "export" CSS to it.`;
+    },
+  },
+};
+const l10n = getIntlData(localizationStrings);
+
+export function run(conf) {
+  if (!conf.lint?.[ruleName]) return;
+  const logger = conf.lint[ruleName] === "warn" ? showWarning : showError;
```

Comment:
Yeah, I guess we can then control it via w3c/defaults at the W3C level.... I'd still like to make this into an error globally, instead of a warning. I think BS treats them also as fatal.